nntrainer/src/tensor.cpp

```diff
@@ -411,8 +411,8 @@ Tensor Tensor::sum(int axis, float alpha) const {
 
   const float *data = getData();
 
-  if (axis >= 4)
-    throw std::out_of_range("Error: Dimension cannot exceed 3");
+  if (axis < 0 || axis >= 4)
```

Comment:
change axis to unsigned int than adding the check of `axis < 0`.

Reply:
Also, can axis be equal to 0 and 4 both valid? out

Reply:
> change axis to unsigned int than adding the check of axis < 0.
Changed to unsigned int. 

> Also, can axis be equal to 0 and 4 both valid? out
Do you mean axis 0 and 4 are both 0? If that's the case semantics for the axis has to be changed. I think it'd better be done in a different patch.

Reply:
> Also, can axis be equal to 0 and 4 both valid? out

I meant that there are 4 dimensions. However,  current check allows for 5 values on axis [0, 4].
I think it should be [0, 4)